mep/pages/models.py

```diff
@@ -51,6 +53,38 @@ class Meta:
         label = 'SVG'
 
 
+class SlugStructValue(blocks.StructValue):
+    '''A :class:`~wagtail.core.blocks.StructValue` that enables generating a
+    slug from a title using django's `slugify`, for use as an anchor href.'''
+    # for more on usage of `StructValue`, see:
+    # https://docs.wagtail.io/en/v2.5/topics/streamfield.html#custom-value-class-for-structblock
+
+    @cached_property
+    def slug(self):
+        '''Slugify and return a StructBlock's title, if one is set.'''
+        # NOTE this implementation will compute this on every page load where
+        # a StructBlock that uses SlugStructValue is included.
+        # A different approach might take advantage of wagtail's hooks:
+        # https://docs.wagtail.io/en/v2.3/reference/hooks.html
+        title = self.get('title')
+        if title:
+            return slugify(title)
+
+
+class LinkableSectionBlock(blocks.StructBlock):
+    ''':class:`~wagtail.core.blocks.StructBlock` for a rich text block and an
+    associated `title` that will render as an <h2>. Creates an anchor (<a>)
+    so that the section can be directly linked to using a url fragment.'''
+    title = blocks.CharBlock()
+    body = blocks.RichTextBlock()
+
+    class Meta:
+        icon = 'form'
+        label = 'Linkable Section'
+        template = 'pages/snippets/linkable_section.html'
+        value_class = SlugStructValue # enables value.slug() in template
```

Comment:
Tiniest of quibbles, only flagging because I've seen it in your code before (I'm not sure why your linter doesn't flag it or your auto-pep8 fix it, whatever you use). From [PEP 8 on inline comments](https://www.python.org/dev/peps/pep-0008/#inline-comments):
>  Inline comments should be separated by at least two spaces from the statement.

Reply:
thank you, never knew this! wonder why my linter doesn't flag it, also.